test/integration/npm/publish.test.js

```diff
@@ -51,12 +51,38 @@ describe('npm publish', function () {
       }));
   });
 
-  it('should trigger a carpenter build', function (done) {
+  it('should trigger a carpenter build with promote=true', function (done) {
     sinon.spy(context.registry, 'cacheRequest');
     sinon.spy(context.app.publisher.carpenter, 'build');
 
     macros.publishOk(context)(function () {
       assume(context.app.publisher.carpenter.build).to.be.called();
+      assume(context.app.publisher.carpenter.build).to.be.calledWith(
```

Comment:
you can simplify this with [`calledWithMatch`](https://github.com/terinjokes/assume-sinon#calledwithmatch)

Reply:
@msluther wouldn't it be the same due to nested object stuff? Im a sinon nub so im open to the suggestion if you know what it would be

Reply:
it'd be similar and really just removes the sinon.match() call. it'd be something like:

```js
      assume(context.app.publisher.carpenter.build).to.be.calledWithMatch({
        data: {
          payload: sinon.match.any,
          promote: true
        }
      });
```

Reply:
functionally equivalent though